transport/src/main/java/io/netty/channel/nio/NioEventLoop.java

```diff
@@ -486,10 +551,12 @@ private void processSelectedKeysPlain(Set<SelectionKey> selectedKeys) {
                 }
             }
         }
+        return processed;
     }
 
-    private void processSelectedKeysOptimized(SelectionKey[] selectedKeys) {
-        for (int i = 0;; i ++) {
+    private int processSelectedKeysOptimized(SelectionKey[] selectedKeys) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Consider using varargs for methods or constructors which take an array the last parameter. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=pmd%3AUseVarargs)